changing Input component to OldInput

lulen11 opened the pull request:
hey @rin @davidpaley, in this PR I've refactored it so that we're using the right `Input` components going forward (renamed `labeledInput` to `Input` and depreciated the old `Input` component to be `OldInput` 👈 until we're SURE we can delete it)... I'm just not sure about the validations and would love some help checking over these if you have time? Thank you!

bronz3beard commented on src/lib/components/input/input.js:
```diff
@@ -1,83 +1,77 @@
-          {...restProps}
+          onChange={handleChange}
+          value={value}
+          {...inputProps}
```
I don't know if they need to be or not but the props `...spread` items do they need to be in `proptypes`?

rin replied:
1. The whole thing with the proptypes is … optional, the way we use it, so it's always a nice to have, and nothing breaks if we don't do it 100%. 
2. I think the idea is that we just pass all the rest of the props, whatever they might be. If that is the case, we don't know what's in there. 

So … I'd leave it like that, unless we decide we want to be more strict about things like that.

bronz3beard commented on src/lib/components/input/input.md:
```diff
@@ -1,65 +1,79 @@
+}
+
 <Input
-  value="I am read only. Don't try to change me!"
```
This needs to be on a t-shirt!!!  `value="I am read only. Don't try to change me!"`

bronz3beard commented on src/lib/components/input/input.module.scss:
```diff
@@ -2,76 +2,146 @@
+    text-align: left;
+    transform: translate3d(0, 0, 0) scale(1);
+    transform-origin: left top;
+    transition: transform 0.4s cubic-bezier(0.25, 0.8, 0.25, 1);
```
that's a cool bit of css!

rin commented on src/lib/components/input/input.js:
```diff
@@ -1,83 +1,77 @@
-          className={`${isValidationOk() && styles.error} ${styles.input} ${className}`}
-          value={value}
+          className={`${className} ${error && styles.error}`}
+          type={type || "text"}
```
Rory is right, you can just make that `type={type}`, because further down the default is set to `text` anyway.

rin commented on src/lib/components/input/input.js:
```diff
@@ -1,83 +1,77 @@
-      validationMessage,
-      hasErrorAfterSubmit,
-      // props handed to the input DOM element
-      ...restProps
```
I think I'd like to have this piece …

rin commented on src/lib/components/input/input.js:
```diff
@@ -1,83 +1,77 @@
-import React, { PureComponent } from 'react';
-import PropTypes from 'prop-types';
-import { componentPropTypes, defaultComponentPropTypes } from '../../utils/componentPropTypes';
-import { withValidation } from '../../utils/hocs';
```
… and we will need this to make the validations work …

lulen11 replied:
I've just changed it so that we're pulling the file from the `input` folder rather than the `oldInput` folder which will soon be deprecated. I hope that's ok @rin

rin commented:
Hey Lara! It's already the end of my working day and I didn't get to finish reviewing this, but we'll have to add a few changes to make validations work. (Not much, though!)
I also pushed a tiny commit before otherwise it wouldn't build.

I can make validations work tomorrow and also explain it to you (unless someone can get to it earlier …?). The other thing is, as I already said when trying to use this for the Profile in Portal, it doesn't work for me when I try to use it in Portal … the beautiful animation where the placeholder text becomes the little label on top just doesn't happen – they just look like regular input fields, no matter which theme I choose. Is there any secret to making this work? 😢

lulen11 commented:
😭 they should work... that's so weird... the way we've done it is just through CSS animations but it could be because the label and placeholder aren't being defined in the same way on the portal? @bronz3beard could you please take a look at this & the validations with @rin tonight? 🙏

rin commented:
> 😭 they should work... that's so weird... the way we've done it is just through CSS animations but it could be because the label and placeholder aren't being defined in the same way on the portal? @bronz3beard could you please take a look at this & the validations with @rin tonight? 🙏

it's not unusual really … for example the styles for selects are also not being picked up correctly from Blueprint in Portal, so we always have to overwrite them with styles in Portal. I don't know if it's just something about … CSS specificity or something, or another problem. I will look into the CSS I see for the labels in Portal. Maybe it can be solved by adding lots of `important!`s … 🙈

rin commented:
added another commit to make the validation messages display.

Here's instructions on how to test:
- check out this branch locally
- check out Portal staging branch locally
- in Blueprint, run `yarn build:lib`
- in Blueprint, run `yarn link`
- in Portal, run `yarn link 'aime-blueprint'`
- run Portal (`yarn start`)
- Log in to Portal, go to the Mentor application, check out how the first name field looks; empty the name field and click "Next" – you should see a red validation message.

davidpaley commented on src/lib/components/input/input.js:
```diff
@@ -1,83 +1,98 @@
+  componentPropTypes,
+  defaultComponentPropTypes,
+} from "../../utils/componentPropTypes";
+import { handleInputChange } from "../oldInput/utils.ignore";
```
I don't like to have this oldInput thing here. Maybe it is something that we can add to this new input component if we need it

lulen11 replied:
Yeah for sure. The idea is to only use the `oldInput` files for the immediate future. Once we're sure the refactored `input` is working, we will delete the `oldInput` folder

lulen11 replied:
I've copied in the `utils.ignore.js` file from `oldInput` folder into `input` folder and have adjusted the import call to suit... I hope this is ok @rin and @davidpaley - https://github.com/aimementoring/blueprint/pull/179/commits/b498d9454b5c4d19e8c1d346a43bb92eb47c37a5

lulen11 commented:
hey @davidpaley and @rin I've tried to test locally and ran into this issue which I believe is because we're still using `labeledInput` in a couple of instances. 
<img width="1680" alt="Screen Shot 2020-02-26 at 10 22 37 am" src="https://user-images.githubusercontent.com/39749772/75300136-1426b180-588b-11ea-946e-bbc661935027.png">

So for the purpose of testing, on my local `staging` branch of the portal, I just replaced any instances of `LabeledInput` with `Input` and tried again. I eventually got it working (had some yarn upgrade issues but worked it out i think) and it's looking like they're functioning. There are some things we will need to fix in the portal once importing these changes. ie. When I changed to `Input` on `firstInput.tsx` I got an extra box on the login field. I think this is due to some styling on the invisible error space we have set here. 

<img width="1680" alt="Screen Shot 2020-02-26 at 11 31 26 am" src="https://user-images.githubusercontent.com/39749772/75300385-d0807780-588b-11ea-9ba3-22f43d8c1d1b.png">


And also wanted to show you how the validation message is looking for me when I tried your testing instructions @rin :
<img width="1680" alt="Screen Shot 2020-02-26 at 11 31 01 am" src="https://user-images.githubusercontent.com/39749772/75300423-e726ce80-588b-11ea-98e1-47446f47b05d.png">

It seems to work but maybe there are some console errors we might need to worry about?

davidpaley commented:
@lulen11 I fixed the problem with this branch. The error was related with the validations, it was not made as it was in the other components. Now it is working, I am testing this with staging, using Inputs in everywhere and this work great.

davidpaley commented:
Also with this change, the input type date is broken, look:

![image](https://user-images.githubusercontent.com/4527780/75371802-2feb8100-58a6-11ea-90f0-cd2651f62a65.png)

camposcristianeze commented:
How can I quickly test this in `aime-portal` ?

My idea is.. If we can put it in a "working" and "look's ok" state for everyone on the staff that will test before our meeting tomorrow (internal release) morning then yes!

Otherwise let's do our tests with the old version and before the public release (1 more week?) we should have that sorted. @lulen11 you'll have a clearer idea of what's left there so let us know what you prefer 👍🏼
